### DISCUSSION TITLE
Using something ala Spring Data JDBC with Quarkus.

### DISCUSSION THREAD

**Discussion by bjorntj** [quarkusio/quarkus#38740] [Community]
Hi.

I don't like using ORMs for accessing databases, don't like that I need to fetch data from the database before doing a save, for instance.
But I really do like Quarkus, so I was wondering if it really isn't possible to have database access without using Hibernate or Panache? (without writing old fashion JDBC Java SQL code?

**Comment by luca-bassoricci** [comment]
https://quarkus.io/extensions/io.quarkiverse.jooq/quarkus-jooq/

**Reply by bjorntj** [reply]
Thx.. But I do not like JOOQ either. I would like to write the SQL myself, when it is more than standard CRUD methods, those I would like to just use.. And I know, I am a bit weird.. :)

**Comment by maxandersen** [comment]
You don't need to load data from the database to save it.

Tried to use stateless session?

**Reply by bjorntj** [reply]
No, I haven't. Maybe worth checking out.

**Reply by bjorntj** [reply]
But that does not really solve the issue where I like to write complex SQL queries myself. But might be that the Query annotation can be used.
Really like the new JdbcClient in Spring Boot for writing SQL queries.

**Reply by luca-bassoricci** [reply]
[Apache commons DbUtils](https://commons.apache.org/proper/commons-dbutils/index.html).
But I suppose you can use Hibernate without entities, just use stateless sessions and native queries.

**Reply by bjorntj** [reply]
Seems to be too much manually work at first glance but will check it out. Thx. :)

**Comment by KangoV** [comment]
Although Spring Data JDBC is nice, it does not support composite keys, so is pretty useless for existing databases.

**Comment by sathishk** [comment]
@bjorntj @KangoV 

[SQL Builder](https://github.com/tamilnadujug/sql-builder) is a framework-independent, lightweight alternative to Spring JDBC Client. You can find quarkus example [here](https://github.com/tamilnadujug/sql-builder/tree/develop/examples/sqlbuilder-quarkus)
